HymnsTests/Display/DisplayHymnViewModelSpec.swift

```diff
@@ -5,26 +5,27 @@ import Quick
 @testable import Hymns
 
 class DisplayHymnViewModelSpec: QuickSpec {
-
+    
     override func spec() {
         describe("DisplayHymnViewModel") {
             let testQueue = DispatchQueue(label: "test_queue")
             var hymnsRepository: HymnsRepositoryMock!
             var favoritesStore: FavoritesStoreMock!
             var historyStore: HistoryStoreMock!
             var target: DisplayHymnViewModel!
-            var webViewPreloader: WebViewPreloaderMock!
+            var analytics: AnalyticsLoggerMock!
+            var pdfLoader: PDFLoaderMock!
             beforeEach {
                 hymnsRepository = mock(HymnsRepository.self)
                 favoritesStore = mock(FavoritesStore.self)
                 historyStore = mock(HistoryStore.self)
-                webViewPreloader = mock(WebViewPreloader.self)
+                analytics = mock(AnalyticsLogger.self)
+                pdfLoader = mock(PDFLoader.self)
             }
             describe("fetching hymn") {
                 context("with nil repository result") {
                     beforeEach {
-                        target = DisplayHymnViewModel(backgroundQueue: testQueue, hymnToDisplay: classic1151, hymnsRepository: hymnsRepository,
-                                                      favoritesStore: favoritesStore, historyStore: historyStore)
+                        target = DisplayHymnViewModel(analytics: analytics, backgroundQueue: testQueue, favoritesStore: favoritesStore, hymnToDisplay: classic1151, historyStore: historyStore, pdfPreloader: pdfLoader)
```

Comment:
Two things here:

1) You don't really need to mock the analytics logger because you never make any assertions on it, nor do you mock any of its methods. If you want to include it, then you should put assertions on it.
2) Noticed that there used to be a hymnsRepository: hymnsRepository in the constructor that you removed. That's the source of a lot of your test failures, since you are no longer using the mocked one and you're using the real one.

A big giveaway was this test failing:
```                       
it("should call hymnsRepository.getHymn") {
  verify(hymnsRepository.getHymn(classic1151)).wasCalled(exactly(1))
}
```

Because in fetchHymn, one of the first things we do is repository.getHymn:
```   
 func fetchHymn() {
        analytics.logDisplaySong(hymnIdentifier: identifier)
        repository
            .getHymn(identifier)
  ...
  }
```
So if `hymnsRepository.getHymn` is not being called, that should be a red flag that you're using the right repository.

Reply:
Nice! Thank you!